### PR TITLE
Java 11 compatibility

### PR DESCRIPTION
**Pull request by lucebac**

### CODE REVIEW

**Comment by ic0ns** [comment]
@mgierlings can you take a look at this/these PR's?

**Comment by lucebac** [comment]
especially the changed test in this PR does need review b/c i needed to change the source/expected (java 11 formats the xml differently than java 8). I could use some help here on how to configure the formatter properly.

**Comment by lucebac** [comment]
From what I can tell, the issue with the incorrect formatting is now resolved with the latest commits. Could you please verify that?

File: pom.xml [hunk 1]
```diff
@@ -66,22 +66,38 @@
...
+            <version>2.13.3</version>
         </dependency>
         <dependency>
             <groupId>org.apache.logging.log4j</groupId>
             <artifactId>log4j-core</artifactId>
-            <version>2.13.2</version>
+            <version>2.13.3</version>
+        </dependency>
+        <dependency>
+            <groupId>jakarta.xml.bind</groupId>
```
Comment (jurajsomorovsky) [inline pom.xml:83]: what is the reason for using jakarta and not javax.xml.bind?
https://www.jesperdj.com/2018/09/30/jaxb-on-java-9-10-11-and-beyond/
(javax is also recommended by the official doc)

**Review by jurajsomorovsky — COMMENTED** [review]
Looks good to me, just please clarify my question

**Comment by jurajsomorovsky** [comment]
Btw, I am getting the following error when using OpenJDK 11
```
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.google.inject.internal.cglib.core.$ReflectUtils$1 (file:/usr/share/maven/lib/guice.jar) to method java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain)
WARNING: Please consider reporting this to the maintainers of com.google.inject.internal.cglib.core.$ReflectUtils$1
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
```

Apparently, this will should be fixed with newer versions, so no problem, see https://github.com/google/guice/issues/1133

File: pom.xml (hunk 1 above)
Comment (lucebac) [reply pom.xml]: fixed with 3d917fa

**Comment by lucebac** [comment]
> Btw, I am getting the following error when using OpenJDK 11
> 
> ```
> WARNING: An illegal reflective access operation has occurred
> WARNING: Illegal reflective access by com.google.inject.internal.cglib.core.$ReflectUtils$1 (file:/usr/share/maven/lib/guice.jar) to method java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain)
> WARNING: Please consider reporting this to the maintainers of com.google.inject.internal.cglib.core.$ReflectUtils$1
> WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
> WARNING: All illegal access operations will be denied in a future release
> ```
> 
> Apparently, this will should be fixed with newer versions, so no problem, see [google/guice#1133](https://github.com/google/guice/issues/1133)

I'm not seeing the illegal reflected access you mentioned. Could you tell me, when this happened? Guice is not a direct dependency of ModifiableVariable, nor does it seem to be a transitive dependency (according to `mvn dependency:tree`). Thus, I think you are just using an old version of maven that produces this errors on Java 11.

**Comment by jurajsomorovsky** [comment]
Yes, this seems to be only because of my mvn version:
```
Apache Maven 3.6.3
Maven home: /usr/share/maven
Java version: 14.0.1, vendor: Private Build, runtime: /usr/lib/jvm/java-14-openjdk-amd64
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "5.4.0-51-generic", arch: "amd64", family: "unix"
```

**Review by jurajsomorovsky — APPROVED** [review]

**Pull request merged**